### PR TITLE
Add XML challenge for session 15.

### PR DESCRIPTION
**Pull request by alinaitschool**
- add XML challenge for session 1

Implementation was followed the documentation:
1.https://www.baeldung.com/jackson-object-mapper-tutorial
2. https://www.baeldung.com/java-serialization-approaches

@AlexandruIliescu can you please, approve this PR?

### CODE REVIEW

File: src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/Book.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package com.github.alina.session_17_generics.challenge.xml_to_json;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+
+public class Book {
+    @JacksonXmlProperty(localName = "isbn")
```
Comment (AlexandruIliescu) [inline src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/Book.java:8]: Good use of the library, that's great!

File: src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/BookStore.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.github.alina.session_17_generics.challenge.xml_to_json;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+
+import java.util.List;
+
+public class BookStore {
+    @JacksonXmlElementWrapper(useWrapping = false)
+    private List<Book> book;
```
Comment (AlexandruIliescu) [inline src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/BookStore.java:10]: Please pay attention to the name of the properties. When we use List<>, we can consider names ending with `...List` or the plural of the property.

File: src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/Convertor.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+public class Convertor {
+    public static void main(String[] args) throws IOException {
+        XmlMapper mapper = new XmlMapper();
+        XmlMapper xmlMapper = new XmlMapper();
+        File file = new File("/Users/alina/Desktop/JavaCourseStartSteps/ExerciseAlgoritm/itschool1/src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/library.xml");
+        BookStore bookStore = xmlMapper.readValue(file, BookStore.class);
+        System.out.println(bookStore);
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.writeValue(new File("/Users/alina/Desktop/JavaCourseStartSteps/ExerciseAlgoritm/itschool1/src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/library.json"), bookStore);
```
Comment (AlexandruIliescu) [inline src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/Convertor.java:19]: Please consider to use relative paths when possible. Our application, at least in the most cases, with consume/generate files inside of the project structure, and in this case, our application should not rely or be aware of the OS paths.
You can also make use of Path class:
`Path path = Paths.get("src/main/resources/library.json");`

**Review by AlexandruIliescu — APPROVED** [review]

**Comment by AlexandruIliescu** [comment]
Good job Alina!
Very smooth conversion from XML to JSON.

File: src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/Convertor.java (hunk 3 above)
Comment (alinaitschool) [reply src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/Convertor.java:19]: Hey, was working your suggestion. Thanks.

File: src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/BookStore.java (hunk 2 above)
Comment (alinaitschool) [reply src/main/java/com/github/alina/session_17_generics/challenge/xml_to_json/BookStore.java:10]: Ok, thanks.

**Pull request merged**